### PR TITLE
PS-4464: Expose GTID_EXECUTED for CONSISTENT SNAPSHOT (8.0)

### PR DESCRIPTION
**Pull request by kamil-holubicki**
This patch adds a status variable Binlog_snapshot_gtid_executed with the following changes:

Add tracking of GTID_EXECUTED in Gtid_state::gtid_executed
Add function to snapshot GTID_EXECUTED with Gtid_state::get_snapshot_gtid_executed()
Add a status variable Binlog_snapshot_gtid_executed
Fix/re-record tests where new status is exposed in results.

### CODE REVIEW

File: mysql-test/suite/binlog/t/percona_binlog_snapshot_gtid_executed-master.opt [hunk 1]
```diff
@@ -0,0 +1 @@
+--enforce-gtid-consistency --gtid-mode=ON
```
Comment (percona-ysorokin) [inline mysql-test/suite/binlog/t/percona_binlog_snapshot_gtid_executed-master.opt]: In 8.0 this test should be in `binlog_gtid` test suite.
This .opt file can be removed then.

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

File: mysql-test/suite/binlog/t/percona_binlog_snapshot_gtid_executed-master.opt (hunk 1 above)
Comment (kamil-holubicki) [reply mysql-test/suite/binlog/t/percona_binlog_snapshot_gtid_executed-master.opt]: done

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Comment by kamil-holubicki** [comment]
https://ps80.cd.percona.com/job/percona-server-8.0-param/516/

**Pull request merged**